### PR TITLE
feat: add support for Cloud Build configuration files

### LINKED ISSUE #5502: Google Cloudbuild updating
### Which Renovate are you using?

WhiteSource Renovate App

### Which platform are you using?

GitHub.com

### What would you like to do?

We use google cloud build service which accepts a yaml config, these file uses docker images. We would love to have those images updated too.

Is that something possible? Willing to contribute through PRs if you can point me to where I can get started.

Thank you!

### PR DESCRIPTION
**Pull request by RoopeHakulinen**
## Changes:

This PR adds support for updating Cloud Build Docker image versions.

Closes #5502. 

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/manager/cloudbuild/readme.md [hunk 1]
```diff
@@ -0,0 +1 @@
+The `cloudbuild` manager extracts `docker` datasources from Cloud Build config files.
```
Comment (rarkins) [inline lib/manager/cloudbuild/readme.md:1]: Add a link here to a page which describes the configuration files

File: lib/manager/cloudbuild/index.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { extractPackageFile } from './extract';
+
+export { extractPackageFile };
+
+export const defaultConfig = {
+  fileMatch: [/cloudbuild\.y[a]?ml$/],
```
Comment (rarkins) [inline lib/manager/cloudbuild/index.ts:6]: Should be a string

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/manager/cloudbuild/index.ts (hunk 2 above)
Comment (RoopeHakulinen) [reply lib/manager/cloudbuild/index.ts:6]: Thanks! Is there a standard on how to handle .yml vs .yaml with that approach?

File: lib/manager/cloudbuild/readme.md (hunk 1 above)
Comment (RoopeHakulinen) [reply lib/manager/cloudbuild/readme.md]: Added.

File: lib/manager/cloudbuild/index.ts (hunk 2 above)
Comment (RoopeHakulinen) [reply lib/manager/cloudbuild/index.ts]: Seemed like `fileMatch: ['(^|/)cloudbuild.ya?ml']` might work as intended

Comment (rarkins) [reply lib/manager/cloudbuild/index.ts]: Yes, that would.

File: lib/manager/cloudbuild/extract.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+export function extractPackageFile(content: string): PackageFile {
+  const deps: PackageDependency[] = [];
+  try {
+    const doc = yaml.safeLoad(content) as any;
+    if (doc?.steps && is.array(doc.steps)) {
+      for (const step of doc.steps) {
+        if (step.name) {
+          const dep = getDep(step.name);
+          dep.depType = 'docker';
+          dep.versioning = 'docker';
```
Comment (viceice) [inline lib/manager/cloudbuild/extract.ts:16]:
```suggestion
```

Will be done automagically

File: lib/manager/cloudbuild/extract.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+          logger.debug(
```
Comment (viceice) [inline lib/manager/cloudbuild/extract.ts:17]:
```suggestion
          logger.trace(
```

File: lib/manager/cloudbuild/extract.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+          deps.push(dep);
+        }
+      }
+    }
+  } catch (err) /* istanbul ignore next */ {
+    if (err.stack?.startsWith('YAMLException:')) {
+      logger.debug({ err });
+      logger.debug(
+        'YAML exception extracting Docker images from a Cloud Build configuration file.'
+      );
```
Comment (viceice) [inline lib/manager/cloudbuild/extract.ts:35]:
```suggestion
      logger.debug(
        { err },
        'YAML exception extracting Docker images from a Cloud Build configuration file.'
      );
```

**Review by viceice — CHANGES_REQUESTED** [review]

**Comment by RoopeHakulinen** [comment]
The tests seem to be failing due to the issue shown in the screenshot below. I'm not quite sure why that is since when I run the tests locally with `yarn jest -u lib/manager/cloudbuild/` it doesn't update them at all. Any thoughts on how to solve this?

image

**Review by viceice — APPROVED** [review]
Strage to me to use the step name as image name 😏

**Comment by RoopeHakulinen** [comment]
It is a bit oddly named indeed..

**Pull request merged**